libsquoosh/lib/move-d-ts.js

```diff
@@ -4,23 +4,51 @@ const path = require('path');
 
 const tsOutputDir = path.resolve('..', '.tmp', 'ts', 'libsquoosh');
 const tsOutputSourceDir = path.join(tsOutputDir, 'src');
-const buildDir = path.resolve('build');
+const libSquooshTypesOutputDir = path.resolve('build', 'libsquoosh', 'types');
 
-(async () => {
-  await del(path.join(buildDir, '*.d.ts'));
-  const files = await fs.promises.readdir(tsOutputSourceDir);
+const codecsDir = path.resolve('..', 'codecs');
+const codecsTypesOutputDir = path.resolve('build', 'codecs');
+
+async function getFilesRecursive(from) {
+  const filesOrDirectories = await fs.promises.readdir(from, {
+    withFileTypes: true,
+  });
+
+  const allFiles = await Promise.all(
+    filesOrDirectories.flatMap((fileOrDirectory) => {
+      if (fileOrDirectory.isFile()) {
+        return path.resolve(from, fileOrDirectory.name);
+      }
+
+      return getFilesRecursive(path.resolve(from, fileOrDirectory.name));
```

Comment:
```suggestion
      } else if (
        fileOrDirectory.isDirectory() &&
        !fileOrDirectory.name !== 'node_modules'
      ) {
        return getFilesRecursive(path.resolve(from, fileOrDirectory.name));
      }
      return [];
```

Otherwise it bails out over some symlinks that are created as part of the codec builds

Reply:
Oh makes a lot of sense, thanks for the catch; updated it! :)